src/test/regress/sql/multi_router_planner.sql

```diff
@@ -0,0 +1,402 @@
+INSERT INTO articles_hash VALUES ( 7,  7, 'aseptic', 12298);
+INSERT INTO articles_hash VALUES ( 8,  8, 'agatized', 16368);
+INSERT INTO articles_hash VALUES ( 9,  9, 'alligate', 438);
+INSERT INTO articles_hash VALUES (10, 10, 'aggrandize', 17277);
+INSERT INTO articles_hash VALUES (11,  1, 'alamo', 1347);
+INSERT INTO articles_hash VALUES (12,  2, 'archiblast', 18185);
+INSERT INTO articles_hash VALUES (13,  3, 'aseyev', 2255);
+INSERT INTO articles_hash VALUES (14,  4, 'andesite', 19094);
+INSERT INTO articles_hash VALUES (15,  5, 'adversa', 3164);
+INSERT INTO articles_hash VALUES (16,  6, 'allonym', 2);
+INSERT INTO articles_hash VALUES (17,  7, 'auriga', 4073);
+INSERT INTO articles_hash VALUES (18,  8, 'assembly', 911);
+INSERT INTO articles_hash VALUES (19,  9, 'aubergiste', 4981);
+INSERT INTO articles_hash VALUES (20, 10, 'absentness', 1820);
+INSERT INTO articles_hash VALUES (21,  1, 'arcading', 5890);
+INSERT INTO articles_hash VALUES (22,  2, 'antipope', 2728);
+INSERT INTO articles_hash VALUES (23,  3, 'abhorring', 6799);
+INSERT INTO articles_hash VALUES (24,  4, 'audacious', 3637);
+INSERT INTO articles_hash VALUES (25,  5, 'antehall', 7707);
+INSERT INTO articles_hash VALUES (26,  6, 'abington', 4545);
+INSERT INTO articles_hash VALUES (27,  7, 'arsenous', 8616);
+INSERT INTO articles_hash VALUES (28,  8, 'aerophyte', 5454);
+INSERT INTO articles_hash VALUES (29,  9, 'amateur', 9524);
+INSERT INTO articles_hash VALUES (30, 10, 'andelee', 6363);
+INSERT INTO articles_hash VALUES (31,  1, 'athwartships', 7271);
+INSERT INTO articles_hash VALUES (32,  2, 'amazon', 11342);
+INSERT INTO articles_hash VALUES (33,  3, 'autochrome', 8180);
+INSERT INTO articles_hash VALUES (34,  4, 'amnestied', 12250);
+INSERT INTO articles_hash VALUES (35,  5, 'aminate', 9089);
+INSERT INTO articles_hash VALUES (36,  6, 'ablation', 13159);
+INSERT INTO articles_hash VALUES (37,  7, 'archduchies', 9997);
+INSERT INTO articles_hash VALUES (38,  8, 'anatine', 14067);
+INSERT INTO articles_hash VALUES (39,  9, 'anchises', 10906);
+INSERT INTO articles_hash VALUES (40, 10, 'attemper', 14976);
+INSERT INTO articles_hash VALUES (41,  1, 'aznavour', 11814);
+INSERT INTO articles_hash VALUES (42,  2, 'ausable', 15885);
+INSERT INTO articles_hash VALUES (43,  3, 'affixal', 12723);
+INSERT INTO articles_hash VALUES (44,  4, 'anteport', 16793);
+INSERT INTO articles_hash VALUES (45,  5, 'afrasia', 864);
+INSERT INTO articles_hash VALUES (46,  6, 'atlanta', 17702);
+INSERT INTO articles_hash VALUES (47,  7, 'abeyance', 1772);
+INSERT INTO articles_hash VALUES (48,  8, 'alkylic', 18610);
+INSERT INTO articles_hash VALUES (49,  9, 'anyone', 2681);
+INSERT INTO articles_hash VALUES (50, 10, 'anjanette', 19519);
+
+
+
+SET citus.task_executor_type TO 'real-time';
+SET citus.large_table_shard_count TO 2;
+SET client_min_messages TO 'DEBUG2';
+
+-- insert a single row for the test
+INSERT INTO articles_single_shard_hash VALUES (50, 10, 'anjanette', 19519);
```

Comment:
done.
to be able to support pl/pgsql function, we ended up changing multi_executor so that it has a some cursor support for forward facing ones.